sirepo/package_data/static/js/sirepo.js

```diff
@@ -96,6 +99,58 @@ angular.element(document).ready(function() {
             : addTag(src, 'script', 'body', 'src', {'type': 'text/javascript', 'async': true});
     }
 
+    function loadFromPath(path, type, callback, errCallback, finallyCallback) {
+        let d = $.Deferred();
+        $.get(`${path}${SIREPO.SOURCE_CACHE_KEY}`, function (res) {
+            callback(res);
+        }, type)
+            .fail(function (res) {
+                if (errCallback) {
+                    errCallback(res);
+                }
+            })
+            .always(function (res) {
+                if (finallyCallback) {
+                    finallyCallback(res);
+                }
+                d.resolve();
+            });
+        return d.promise();
+    }
+
+    function loadBlock(b) {
+        let val = null;
+        let status = 500;
+        return loadFromPath(b.path, b.fileType, function (res) {
+            val = res;
+            status = 200;
+        }, function(res) {
+            // turns out "svg" (etc.) is not a valid AJAX type, so we'll get an error.
+            // However we can get a successful read anyway; if so assign the response text
+            status = res.status;
```

Comment:
constrain the test here to `ok = res.status == 200`.  This encapsulates the http part of the implementation.

Reply:
60e7ccc